mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/scaladsl/MqttSession.scala

```diff
@@ -162,8 +165,9 @@ final class ActorMqttClientSession(settings: MqttSessionSettings)(implicit mat:
                 case ClientConnector.ForwardPubRel(packetId) =>
                   PubRel(packetId).encode(ByteString.newBuilder).result()
               }.mapError {
-                case ClientConnector.PingFailed => ActorMqttClientSession.PingFailed
-              })
+                  case ClientConnector.PingFailed => ActorMqttClientSession.PingFailed
+                }
+                .watchTermination()((_, done) => done.foreach(_ => killSwitch.shutdown())))
             )
```

Comment:
This is the crux of the change for the PR. When our connect stream ends it means that the session has moved to a disconnected state. We, therefore, need to kill upstream.